contracts/PodCastNFT.sol

```diff
@@ -145,4 +193,26 @@ contract PodCastNFT is ERC721URIStorage, Ownable {
         _setTokenURI(newItemId, finalTokenUri);
         return newItemId;
     }
+
+    function mintWithHenkaku(
+        string memory _imageURI,
+        string memory _name,
+        uint256 _amount
+    ) public onlyValidData(_imageURI, _name) onlyNoneHolder(msg.sender) {
+        require(
+            _amount >= price,
+            "Low amount. you cannot buy with such amount"
+        );
```

Comment:
This implementation is valid for large amount of henkaku, but it may be better to check exactly the same as the price.

```suggestion
        require(
            _amount == price,
            "Not Enough Henkaku"
        );
```

Reply:
it's kinda fun to make it greater lol 
no one tries to put more than it needs to be.
but I saw many implementation which is doing exactly the same